cpp/memilio/mobility/graph.h

```diff
@@ -353,22 +356,26 @@ IOResult<void> set_edges(const fs::path& data_dir, Graph<Model, MigrationParams>
             // mobility coefficients have the same number of components as the contact matrices.
             // so that the same NPIs/dampings can be used for both (e.g. more home office => fewer commuters)
             auto mobility_coeffs = MigrationCoefficientGroup(contact_locations_size, populations.numel());
-
+            auto num_age_groups  = (size_t)params_graph.nodes()[county_idx_i].property.parameters.get_num_groups();
+            commuting_weights =
+                (commuting_weights.size() == 0 ? std::vector<ScalarType>(num_age_groups, 1.0) : commuting_weights);
             //commuters
             auto working_population = 0.0;
-            auto min_commuter_age   = AgeGroup(2);
-            auto max_commuter_age   = AgeGroup(4); //this group is partially retired, only partially commutes
-            for (auto age = min_commuter_age; age <= max_commuter_age; ++age) {
-                working_population += populations.get_group_total(age) * (age == max_commuter_age ? 0.33 : 1.0);
+            int age_index           = 0;
+            for (auto age = AgeGroup(0); age < populations.template size<mio::AgeGroup>(); ++age) {
+                working_population += populations.get_group_total(age) * commuting_weights[age_index];
```

Comment:
```suggestion
                working_population += populations.get_group_total(age) * commuting_weights[size_t(age)];
```

Should `size_t(age)` not work? Do you need another variable?